testsuite/integration-arquillian/test-apps/servlets/src/main/java/org/keycloak/testsuite/adapter/servlet/LinkAndExchangeServlet.java

```diff
@@ -73,42 +75,29 @@ public AccessTokenResponse doTokenExchange(String realm, String token, String re
             String exchangeUrl = KeycloakUriBuilder.fromUri(ServletTestUtils.getAuthServerUrlBase())
                     .path("/auth/realms/{realm}/protocol/openid-connect/token").build(realm).toString();
 
-            URL url = new URL(exchangeUrl);
-            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-            conn.setRequestMethod("POST");
-            conn.setDoInput(true);
-            conn.setDoOutput(true);
+            HttpClient client = new HttpClientBuilder().disableTrustManager().build();
```

Comment:
Will be again good to close the client. Possibly with the auto-closeable block. See the other comment.